File: examples/node-react/backend/src/services/UsersService.ts
import { IUsersService } from "@veld/generated/interfaces/IUsersService";
import { User, CreateUserInput, SuccessResponse } from "@veld/generated/types/users";
import { usersErrors } from "@veld/generated/errors/users.errors";
import { randomUUID } from "crypto";

const store: User[] = [
  { id: "1", name: "Alice", email: "alice@example.com" },
  { id: "2", name: "Bob",   email: "bob@example.com" },
];

export class UsersService implements IUsersService {
  async listUsers(): Promise<User[]> {
    return store;
  }

  async getUser(id: string): Promise<User> {
    const user = store.find((u) => u.id === id);

    if (!user) throw usersErrors.getUser.notFound(`User ${id} not found`);
    return user;
  }

  async createUser(input: CreateUserInput): Promise<User> {
    const exists = store.find((u) => u.email === input.email);
    if (exists) throw usersErrors.createUser.userExists(`Email ${input.email} already exists`);
    if (!input.name || !input.email) throw usersErrors.createUser.conflict("Name and email required");
    const user: User = { id: randomUUID(), ...input };
    store.push(user);
    return user;
  }

  async deleteUser(id: string): Promise<SuccessResponse> {
    const idx = store.findIndex((u) => u.id === id);
    if (idx === -1) throw usersErrors.deleteUser.notFound(`User ${id} not found`);
    store.splice(idx, 1);
    return { status: true };
  }
}

File: examples/python-vue/backend/services/users_service.py
import uuid
from generated.interfaces.i_users_service import IUsersService

_store = [
    {"id": "1", "name": "Alice", "email": "alice@example.com"},
    {"id": "2", "name": "Bob",   "email": "bob@example.com"},
]


class UsersService(IUsersService):
    def list_users(self):
        return list(_store)

    def get_user(self, id: str):
        user = next((u for u in _store if u["id"] == id), None)
        if user is None:
            raise LookupError(f"User {id} not found")
        return user

    def create_user(self, input):
        user = {"id": str(uuid.uuid4()), **input}
        _store.append(user)
        return user

    def delete_user(self, id: str):
        idx = next((i for i, u in enumerate(_store) if u["id"] == id), None)
        if idx is None:
            raise LookupError(f"User {id} not found")
        _store.pop(idx)
